core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -538,6 +557,14 @@ impl VoteStorage {
 
         scanner.finalize().payload.reached_end_of_slot
     }
+
+    fn cache_epoch_boundary_info(&mut self, bank: &Bank) {
+        if matches!(self.vote_source, VoteSource::Gossip) {
+            panic!("Gossip vote thread should not be checking epoch boundary");
```

Comment:
Do we need to panic here? Could we just log an error and early return?

Reply:
I prefer a panic here because it indicates programmer error, only one thread should ever attempt to take the write lock.